### PR TITLE
sile: vendor resources

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Don't merge this as-is, obviously. Ref: https://github.com/Homebrew/brew/pull/2473

### CODE REVIEW

File: Formula/sile.rb [hunk 1]
```diff
@@ -34,6 +63,17 @@ def install
                           "--prefix=#{prefix}"
     system "make"
     system "make", "install"
+    # There's a handful of broken dylibs reported to old `icu4c` versions
+    # and such. This seems to be resolved in HEAD. Not sure what's going on
```
Comment (ilovezfs) [inline Formula/sile.rb]: Did you figure out how they fixed it?

Comment (DomT4) [inline Formula/sile.rb]: No, not yet. It doesn't help that between the last release & the current master there's been a _ton_ of commits including some major rewriting & I haven't fancied bisecting the whole party between the last release & now yet 😢.

There's still `lua-zlib`'s ridiculous behaviour to report upstream to `lua-zlib` as well.

**Comment by DomT4** [comment]
@probot-stale[bot] Relax bot, I haven't forgotten 😉.

**Comment by ilovezfs** [comment]
@DomT4 any news here?

**Comment by DomT4** [comment]
I need to ask upstream for a release. It would simplify this formula & also fix the existing breakage which I'm having trouble tracing to a source. I also need to politely ask `lua-zlib` what on earth is going on with the way they package their rock.

**Comment by DomT4** [comment]
Ref: https://github.com/simoncozens/sile/issues/482

**Comment by alerque** [comment]
@DomT4 Looking into getting a release done upstream...

File: Formula/sile.rb [hunk 2]
```diff
@@ -12,20 +12,49 @@ class Sile < Formula
...
+  end
+
+  resource "luafilesystem" do
+    url "https://github.com/keplerproject/luafilesystem/archive/v_1_6_3.tar.gz"
+    sha256 "5525d2b8ec7774865629a6a29c2f94cb0f7e6787987bf54cd37e011bfb642068"
+  end
+
+  # For some reason it insists on doing another git clone at build because of
+  # the contents of the rockspec. This git clone is done over the git://
+  # protocol because life is bad. This should be reported upstream.
```
Comment (ilovezfs) [inline Formula/sile.rb]: You can use `luarocks make` to get it to use an already staged source tree.

Comment (DomT4) [reply Formula/sile.rb]: 👍. I'd kind of wedged this PR on hold waiting for an upstream release, but I'll remember to implement that one that way if we ever get there.

Comment (ilovezfs) [reply Formula/sile.rb]:
```
Josephs-MacBook-Pro:~ joe$ brew formula-analytics sile
install events in the last 30 days for sile
=====================================================================================================
1 | sile --HEAD                                                                         | 7 | 100.00%
=====================================================================================================
Total                                                                                   | 7 |    100%
=====================================================================================================
Josephs-MacBook-Pro:~ joe$ 
```
Maybe just pick a revision and call it a day?

Comment (DomT4) [reply Formula/sile.rb]: Just to check I'm not chasing you up the wrong tree here: You're happy for me to simply use the latest upstream commit of `sile` pending a new release to get this done?

Comment (ilovezfs) [reply Formula/sile.rb]: Yup. Every install in the last 30 days was --HEAD and 1/3 over the last 100 days. So I don't think we need to stand on ceremony at this point.

Comment (DomT4) [reply Formula/sile.rb]: Cool, I'll finish this one off this afternoon 👍.

Comment (ilovezfs) [reply Formula/sile.rb]: thanks :)

**Comment by DomT4** [comment]
Not pretty but ready for review.

File: Formula/sile.rb [hunk 3]
```diff
@@ -21,22 +19,62 @@ class Sile < Formula
...
     system "./configure", "--disable-debug",
                           "--disable-dependency-tracking",
                           "--disable-silent-rules",
                           "--with-lua=#{prefix}",
                           "--prefix=#{prefix}"
     system "make"
     system "make", "install"
+
+    (libexec/"bin").install bin/"sile"
+    (bin/"sile").write <<-EOS.undent
```
Comment (ilovezfs) [inline Formula/sile.rb]: ~

Comment (DomT4) [reply Formula/sile.rb]: Oh, yeah 🙈. Forgot about that briefly.

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
s :isle_of_man:

**Comment by DomT4** [comment]
Here's hoping for smooth sailing in the wild. Thanks for merging 🙇.

**Pull request closed without merging**